Modules/Filtering/ImageStatistics/include/itkStatisticsImageFilter.h

```diff
@@ -163,27 +133,30 @@ class ITK_TEMPLATE_EXPORT StatisticsImageFilter:
   ~StatisticsImageFilter() override = default;
   void PrintSelf(std::ostream & os, Indent indent) const override;
 
-  /** Pass the input through unmodified. Do this by Grafting in the
-   *  AllocateOutputs method.
-   */
-  void AllocateOutputs() override;
-
   /** Initialize some accumulators before the threads run. */
   void BeforeStreamedGenerateData() override;
 
-  /** Do final mean and variance computation from data accumulated in threads.
+  /** Set outputs to computed values from all regions
    */
   void AfterStreamedGenerateData() override;
 
   void ThreadedStreamedGenerateData( const RegionType &) override;
 
+  itkSetDecoratedOutputMacro(Minimum, PixelType);
+  itkSetDecoratedOutputMacro(Maximum, PixelType);
+  itkSetDecoratedOutputMacro(Mean, RealType);
+  itkSetDecoratedOutputMacro(Sigma, RealType);
+  itkSetDecoratedOutputMacro(Variance, RealType);
+  itkSetDecoratedOutputMacro(Sum, RealType);
+  itkSetDecoratedOutputMacro(SumOfSquares, RealType);
+
 private:
-  CompensatedSummation<RealType> m_ThreadSum;
-  CompensatedSummation<RealType> m_SumOfSquares;
+  CompensatedSummation<RealType> m_ThreadSum{1};
```

Comment:
shouldn't these two be initialized to 0?

Reply:
The value doesn't matter much as it get's properly initialized int BeforeStreamedGenerateData. These were the values in the constructor.